specification/recoveryservices/Recoveryservices.Management/Vault.tsp

```diff
@@ -0,0 +1,321 @@
+      {
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...ResourceGroupParameter,
+      },
+      {
+        /**
+         *The name of the recovery services vault.
+         */
+        @segment("vaults")
+        @path
+        vaultName: string,
+
+        @path
+        @segment("operationStatus")
+        operationId: string,
+      }
+    > {}
+
+interface VaultDeleteOps
+  extends Azure.ResourceManager.Legacy.LegacyOperations<
+      {
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...ResourceGroupParameter,
+      },
+      {
+        /**
+         *The name of the recovery services vault.
+         */
+        @segment("vaults")
+        @path
+        vaultName: string,
+
+        /**
+         * Name of the protection container to unregister.
+         */
+        @path
+        @segment("registeredIdentities")
+        identityName: string,
+      }
+    > {}
+interface Vaults {
+  /**
+   * Get the Vault details.
+   */
+  get is VaultOps.Read<Vault, ErrorType = CloudError>;
+
+  /**
+   * Creates or updates a Recovery Services vault.
+   */
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{vaultName}")
+  createOrUpdate is VaultCreateOps.CreateOrUpdateAsync<
+    Vault,
+    Parameters = {
+      #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+      @header
+      `x-ms-authorization-auxiliary`?: string;
+    },
+    LroHeaders = ArmLroLocationHeader &
+      Azure.Core.Foundations.RetryAfterHeader,
+    ErrorType = CloudError
+  >;
+
+  /**
+   * Updates the vault.
+   */
+  @patch(#{ implicitOptionality: false })
+  update is VaultOps.CustomPatchAsync<
+    Vault,
+    PatchModel = PatchVault,
+    Parameters = {
+      #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+      @header
+      `x-ms-authorization-auxiliary`?: string;
+    },
+    ErrorType = CloudError
+  >;
+
+  /**
+   * Deletes a vault.
+   */
+  delete is VaultOps.DeleteWithoutOkAsync<
+    Vault,
+    Response = ArmAcceptedLroResponse | ArmDeletedNoContentResponse
+  >;
+
+  /**
+   * Fetches all the resources of the specified type in the subscription.
+   */
+  listBySubscriptionId is ArmListBySubscription<
+    Vault,
+    Response = ArmResponse<VaultList>,
+    Error = CloudError
+  >;
+
+  /**
+   * Uploads a certificate for a resource.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @put
+  @operationId("VaultCertificates_Create")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{vaultName}/certificates/{certificateName}")
+  create is VaultOperationOps.ActionNoAutoSync<
+    Vault,
+    CertificateRequest,
+    ArmResponse<VaultCertificateResponse>
+  >;
```

Comment:
I'd think this be in interface `VaultCertificates`, as this is in collection `certificates`, not `vaults`.